speculation-rules.bs

```diff
@@ -627,6 +629,96 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
   1. [=Assert=]: This step is not reached.
 </div>
 
+<h2 id="content-security-policy">Content Security Policy</h2>
+Inline script use is restricted by [[CSP]], but there are cases where a developer doesn't want to permit general inline scripts, but wants to permit inline speculation rules. So, we make the following change that extends [[CSP]] to support a new source keyword for inline speculation rules.
+
+<h3 id="content-security-policy-patches-directives">Directives</h3>
+In the [[CSP#framework-directives]], we define several associated algorithms, and an <dfn grammar noexport>inline check</dfn> algorithm needs a patch to define how to choose the type argument.
+
+  3.  An <dfn for="directive" noexport>inline check</dfn>, which takes an {{Element}}, a
+      <dfn>`type`</dfn> string, a policy, and a source string as arguments,
+      and is executed during [[CSP#should-block-inline]] and during
+      [[CSP#should-block-navigation-request]] for `javascript:` requests. This
+      algorithm returns "`Allowed`" unless otherwise specified.
+      <ins>The <a>`type`</a> string is set to
```

Comment:
This kinda makes an assertion about the calling algorithm, which isn't true without further patches. Supplying "`script attribute`" happens in the code for [event handler content attributes](https://html.spec.whatwg.org/multipage/webappapis.html#event-handler-content-attributes), but nothing in this spec yet invokes the inline check algorithm with the "`script speculationrules`" type.

Presumably this requires a change to the "prepare the script element" patch to update step 19 (which invokes "Should element's inline behavior be blocked by Content Security Security Policy?") to supply "`script speculationrules`" instead of "`script`" in appropriate cases.

Reply:
I think the `event handler content attributes` is the spec for inline javascripts in event handlers, such as &lt;button onlick="run();"&gt;. So it would not be a speculationrules. That's why I skipped to patch it for handling "`script speculationrules`" type.

Reply:
Also, the spec you pointed as the step 19 is the one in the algorithm below?
https://html.spec.whatwg.org/multipage/scripting.html#prepare-the-script-element

So, I agreed that I need to add breakdown steps that setup the `type` and call the algorithm with it to return when it returns "Blocked".

I will update my PR to have a patch to the scripting spec.

Reply:
Ah, OK. Our spec already have a patch for the "prepare the script element".
So, what I should do is to modify the existing patch to be aligned.

Reply:
Now the latest patch contains the mod for the existing patch section.